metadataStore/odm_templates.py

```diff
@@ -26,79 +27,111 @@ class BeginRunEvent(Document):
 
     Attributes
     ----------
+    uid: str
+        Globally unique id for this run
     time : timestamp
-        The date/time as found at the client side when an event is
+        The date/time as found at the client side when an begin_run_event is
         created.
-    beamline_id: str
+    beamline_id : str
         Beamline String identifier. Not unique, just an indicator of
         beamline code for multiple beamline systems
-    owner: str, optional
+    project : str, optional
+        Name of project that this run is part of
+    owner : str
         Specifies the unix user credentials of the user creating the entry
-    scan_id : int, optional
+    group : str
+        Unix group to associate this data with
+    scan_id : int
          scan identifier visible to the user and data analysis
-    beamline_config: bson.ObjectId, optional
+    beamline_config: bson.ObjectId
         Foreign key to beamline config corresponding to a given run
-    custom: dict, optional
-        Additional parameters that data acquisition code/user wants to
-        append to a given header. Name/value pairs
+    sample : dict
+        Information about the sample, may be a UID to another collection
     """
+    uid = StringField(required=True, unique=True)
     time = FloatField(required=True)
-    time_as_datetime = DateTimeField()
+    time_as_datetime = DateTimeField(required=False)
+    project = StringField(required=False)
     beamline_id = StringField(max_length=20, unique=False, required=True)
-    scan_id = StringField(required=False, unique=False)
+    scan_id = IntField(required=True)
     beamline_config = ReferenceField(BeamlineConfig, reverse_delete_rule=DENY,
-                                     required=False,
+                                     required=True,
                                      db_field='beamline_config_id')
```

Comment:
According to the documentation it is...

{
    "description": "Document created at the start of run.  Provides a seach target and later documents link to it",
    "properties": {
        "beamline_config": {
            "description": "Meta-data regrading how the beamline is configured.  Maybe UID into another collection",
            "type": [
                "object",
                "string"
            ]
        },
        "beamline_id": {
            "description": "The beamline ID",
            "type": "string"
        },
        "group": {
            "description": "Unix group to associate this data with",
            "type": [
                "string",
                "integer"
            ]
        },
        "owner": {
            "description": "Unix owner to associate this data with",
            "type": [
                "string",
                "integer"
            ]
        },
        "project": {
            "description": "Name of project that this run is part of",
            "type": "string"
        },
        "sample": {
            "description": "Information about the sample, may be a UID to another collection",
            "type": [
                "object",
                "string"
            ]
        },
        "scan_id": {
            "description": "Scan ID number, not globally unique",
            "type": "integer"
        },
        "time": {
            "description": "Time the run started.  Unix epoch time",
            "type": "number"
        },
        "uid": {
            "description": "Globally unique ID for tihs run",
            "type": "string"
        }
    },
    "required": [
        "uid",
        "time",
        "group",
        "owner",
        "beamline_config",
        "beamline_id"
    ],
    "type": "object"
}